code-challenges.test.js

```diff
@@ -12,41 +14,134 @@
 
 // --------------------1) Create a function that takes in a string and returns a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
 
+// Psuedo Code:
+// Takes in a string as a parameter.
+// I will define the mapping of the object.
+// call the function with each variable as an argument.
+// Iterate over each charate in the string.
+// return the encoded message as a string.
+
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
+const secretCodeWord1 = "Lackadaisical";
 // Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
+const secretCodeWord2 = "Gobbledygook";
 // Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
+const secretCodeWord3 = "Eccentric";
 // Expected output: "3cc3ntr1c"
 
 // b) Create the function that makes the test pass.
 
+describe("encodedWords", () => {
+  const secretCodeWords1 = "Lackadaisical";
+  const secretCodeWords2 = "Gobbledygook";
+  const secretCodeWords3 = "Eccentric";
+
+  it("should convert 'a' to 4, 'e' to 3, 'I' to 1, and 'o' to 0", () => {
+    // expect(encodedWords(secretCodeWords1)).toEqual("L4ck4d41s1c4l");
+    // expect(encodedWords(secretCodeWords2)).toEqual("G0bbl3dyg00k");
+    // expect(encodedWords(secretCodeWords3)).toEqual("3cc3ntr1c");
+  });
+  it("should handle empty strings", () => {});
+});
+
+const hiddenMessage = (message) => {
+  message = message.replace(/a/g, "4");
```

Comment:
I think you have a good start to work from. You can definitely use the replace method for this challenge. This approach is a bit different than the pseudocode, I would consider updating that and making sure you understand how replace works, as well as the regex you are using.